api/envoy/extensions/filters/http/transform/v3/transform.proto

```diff
@@ -0,0 +1,69 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.transform.v3;
+
+import "envoy/config/common/mutation_rules/v3/mutation_rules.proto";
+import "envoy/config/core/v3/substitution_format_string.proto";
+
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.transform.v3";
+option java_outer_classname = "TransformProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/transform/v3;transformv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Transform filter configuration]
+// Transform filter :ref:`configuration overview <config_http_filters_transform>` to perform
+// HTTP header and body transformations.
+// [#extension: envoy.filters.http.transform]
+
+message Transform {
+  // The header mutations to perform.
+  // The :ref:`substitution format specifier <config_access_log_format>` could be applied here.
+  // And except the commonly used format specifiers, there are some additional format specifiers
+  // provided by the transform filter:
+  //
+  // * ``%RQ_BODY(X)%``: the request body. And ``X`` is the JSON pointer to extract a specific
+  //   field from the JSON body. If ``X`` is empty, the whole body will be used.
+  // * ``%RS_BODY(X)%``: the response body. And ``X`` is the JSON pointer to extract a specific
+  //   field from the JSON body. If ``X`` is empty, the whole body will be used.
```

Comment:
> the whole body will be used

FWIW this might cause the headermap to go over the limit, and may be an attack vector, but as this filter doesn't have a security posture, it might be ok.

Also it would be good to clarify whether using this filter implies that all requests/responses are buffered, even if headers_mutations are defined without referencing RQ/RS_BODY. If there's a need to support both buffering and streaming modes, consider adding an enum field specifying the mode (and maybe only support one for now).